File: acm/thai/2012/g.py
for test in range(int(input())):
    input()
    foods = [int(food) for food in input().split()]

    day = 0
    while all(foods):
        foods.sort(reverse=True)
        fridge = foods.pop()
        foods = [food-1 for food in foods]
        foods.append(fridge)
        day += 1

    print(day)

File: acm/thai/2012/g.hs
import Data.List
import Data.List.Split

fridge f = (head sf):(map (\ x -> x - 1) (tail sf)) where sf = sort f

storeFood d f = do
    if all (> 0) f
        then storeFood (d+1) (fridge f)
        else d


test_case = do
    getLine
    foods <- getLine
    let output = storeFood 0 [read x + 0 | x <- splitOn " " foods, x /= ""]
    print output

main = do
    n <- getLine
    sequence_ [test_case | _ <- [1..read n]]
